### PR TITLE
Fn utilities 1.0.7

### PR DESCRIPTION
**Pull request by devsuds**
Changes in function utilities_attachment_to_base64.py and addition of rules and workflows.

### CODE REVIEW

File: fn_utilities/fn_utilities/components/utilities_attachment_to_base64.py [hunk 1]
```diff
@@ -23,35 +23,53 @@ def _attachment_to_base64_function(self, event, *args, **kwargs):
...
 
             log.info("incident_id: %s", incident_id)
             log.info("task_id: %s", task_id)
             log.info("attachment_id: %s", attachment_id)
+            log.info("artifact_id: %s", artifact_id)
+            
             if incident_id is None and task_id is None:
                 raise FunctionError("Error: incident_id or task_id must be specified.")
-            if attachment_id is None:
+            if artifact_id is None and attachment_id is None:
```
Comment (mscherfling) [inline fn_utilities/fn_utilities/components/utilities_attachment_to_base64.py]: I don't believe this is correct to use 'and' logic'. An attachment can be one of three types:
- incident level attachment
- artifact (of type other file)
- task level attachment

so artifact_id and attachment_id are mutually exclusive. task_id and artifact_id are mutually exclusive

Comment (devsuds) [reply fn_utilities/fn_utilities/components/utilities_attachment_to_base64.py]: I understand the confusion, but had to choose this logic since attachment type artifact(other file and log file) has no attribute called attachment in pre-processing script.

File: fn_utilities/fn_utilities/util/customize.py [hunk 2]
```diff
@@ -51,7 +51,6 @@ def customization_data(client=None):
     #     resilient_search_template
     #     resilient_url
     #     rest_body
-    #     rest_cookies
```
Comment (sjcurtinibm) [inline fn_utilities/fn_utilities/util/customize.py]: I don't think this should be removed? Was this input deleted by mistake?

File: fn_utilities/fn_utilities/components/utilities_attachment_to_base64.py [hunk 3]
```diff
@@ -23,35 +23,55 @@ def _attachment_to_base64_function(self, event, *args, **kwargs):
...
+                metadata = client.get(metadata_uri)
             data = client.get_content(data_uri)
 
             results = {
                 "filename": metadata["name"],
                 "content_type": metadata["content_type"],
                 "size": metadata["size"],
                 "created": metadata["created"],
-                "content": base64.b64encode(data)
+                "content": str(base64.b64encode(data))
```
Comment (sjcurtinibm) [inline fn_utilities/fn_utilities/components/utilities_attachment_to_base64.py]: Why do we need to add in this `str()` casting?

File: fn_utilities/fn_utilities/util/customize.py (hunk 2 above)
Comment (devsuds) [reply fn_utilities/fn_utilities/util/customize.py]: Yes, removed by mistake, not really sure though how has it got deleted.

Comment (sjcurtinibm) [reply fn_utilities/fn_utilities/util/customize.py]: If the function that uses this input was modified in the UI, it may have been deleted.

Does regenerating a customize.py file fix the issue?

File: fn_utilities/fn_utilities/components/utilities_attachment_to_base64.py (hunk 3 above)
Comment (devsuds) [reply fn_utilities/fn_utilities/components/utilities_attachment_to_base64.py]: Without casting Base64 object to string if we send results json then it throws a serialization error.

File: fn_utilities/fn_utilities/util/customize.py (hunk 2 above)
Comment (devsuds) [reply fn_utilities/fn_utilities/util/customize.py]: Ideally it does, and just FYI that it is regenerated only. Next time I'll check again.

File: fn_utilities/fn_utilities/components/utilities_attachment_to_base64.py (hunk 3 above)
Comment (sjcurtinibm) [reply fn_utilities/fn_utilities/components/utilities_attachment_to_base64.py]: OK. Has it been tested in Python2.7 and Python3.6? I just wonder why we did not have to cast it before this.

Also make sure that we test this with attachments that have text that contain unicode. You could create a text file containing some characters from this site: https://www.ltg.ed.ac.uk/~richard/unicode-sample-3-2.html and ensure the function works on it

Comment (devsuds) [reply fn_utilities/fn_utilities/components/utilities_attachment_to_base64.py]: ok

File: fn_utilities/fn_utilities/util/customize.py (hunk 2 above)
Comment (devsuds) [reply fn_utilities/fn_utilities/util/customize.py]: after regenerate rest_cookies is still missing

**Comment by sjcurtinibm** [comment]
This is now fn_utilities v1.0.9

**Comment by sjcurtinibm** [comment]
Closing this PR as being handled internally as INT-845
